sdk/nodejs/introspector/test/analysis.spec.ts

```diff
@@ -0,0 +1,331 @@
+import assert from "assert"
+import * as path from "path"
+import { fileURLToPath } from "url"
+
+import { analysis } from "../analysis/analysis.js"
+import { Metadata } from "../analysis/metadata.js"
+import { listFiles } from "../utils/files.js"
+
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+const rootDirectory = `${__dirname}/testdata`
+
+describe("Analysis static Typescript", function () {
+  it("Should correctly analyse a basic class with one method", async function () {
+    const files = await listFiles(`${rootDirectory}/helloWorld`)
+
+    const result = analysis(files)
+    const expected: Metadata = {
+      classes: [
+        {
+          name: "HelloWorld",
+          doc: "HelloWorld class",
+          properties: [],
+          methods: [
+            {
+              name: "helloWorld",
+              returnType: "string",
+              doc: "",
+              params: [
+                {
+                  name: "name",
+                  typeName: "string",
+                  doc: "",
+                  optional: false,
+                },
+              ],
+            },
+          ],
+        },
+      ],
+      functions: [],
+    }
+
+    assert.deepEqual(result, expected)
+  })
+
+  it("Should ignore class that does not have the object decorator", async function () {
+    const files = await listFiles(`${rootDirectory}/noDecorators`)
+
+    const result = analysis(files)
+    const expected: Metadata = {
+      classes: [],
+      functions: [],
+    }
+
+    assert.deepEqual(result, expected)
+  })
+
+  it("Should supports multiple files and classes that returns classes", async function () {
+    const files = await listFiles(`${rootDirectory}/multipleObjects`)
+
+    const result = analysis(files)
+    const expected: Metadata = {
+      classes: [
+        {
+          name: "Bar",
+          doc: "Bar class",
+          properties: [],
+          methods: [
+            {
+              name: "exec",
+              doc: "Execute the command and return its result",
+              returnType: "string",
+              params: [
+                {
+                  name: "cmd",
+                  typeName: "string[]",
+                  doc: "Command to execute",
+                  optional: false,
+                },
+              ],
+            },
+          ],
+        },
+        {
+          name: "Foo",
+          doc: "Foo class",
+          properties: [],
+          methods: [
+            {
+              name: "bar",
+              doc: "Return Bar object",
+              returnType: "Bar",
+              params: [],
+            },
+          ],
+        },
+      ],
+      functions: [],
```

Comment:
What's this, top-level functions? I see no test/example for it.

Reply:
It's for top level function, it will not be supported by the first version but the field exist for a later implementation